sql/sys_vars.cc

```diff
@@ -7391,3 +7391,13 @@ static Sys_var_enum Sys_block_encryption_mode(
   "AES_ENCRYPT() and AES_DECRYPT() functions",
   SESSION_VAR(block_encryption_mode), CMD_LINE(REQUIRED_ARG),
   block_encryption_mode_values, DEFAULT(0));
+
+extern ulonglong opt_binlog_commit_by_rotate_threshold;
+static Sys_var_ulonglong Sys_binlog_free_flush_threshold(
+    "binlog_large_commit_threshold",
+    "When the size of the commiting transaction's binlog events is larger than "
+    "the threshold, try to commit the transaction in the way moving the "
+    "binlog cache file to a binlog file instead of copying the binlog events",
```

Comment:
I think this description can be improved given the name of the variable is now more vague. I suggest something along the lines of
```
Increases transaction concurrency for large transactions (i.e. 
those who's size is larger than this value) by using the large
transaction's cache file as a new binary log, and rotating the
active binary log to the large transaction's cache file at commit
time. This avoids the default commit logic that copies the
transaction cache data to the end of the active binary log file
while holding a lock that prevents other transactions from
binlogging.
```

Reply:
fixed